apps/aecore/test/aecore_suite_utils.erl

```diff
@@ -198,46 +198,63 @@ mine_blocks(Node, NumBlocksToMine, MiningRate, Type) ->
             erlang:error(Reason)
     end.
 
-mine_blocks_until(Node, ConditionFun, Max) ->
-    mine_blocks_until(Node, ConditionFun, 100, Max).
 
-mine_blocks_until(Node, ConditionFun, MiningRate, Max) ->
+mine_blocks_until_tx_on_chain(Node, TxHash, MaxBlocks) ->
+    mine_blocks_until_txs_on_chain(Node, [TxHash], 100, MaxBlocks).
+
+mine_blocks_until_txs_on_chain(Node, TxHashes, MaxBlocks) ->
+    mine_blocks_until_txs_on_chain(Node, TxHashes, 100, MaxBlocks).
+
+mine_blocks_until_txs_on_chain(Node, TxHashes, MiningRate, Max) ->
     ok = rpc:call(
            Node, application, set_env, [aecore, expected_mine_rate, MiningRate],
            5000),
     aecore_suite_utils:subscribe(Node, block_created),
-    aecore_suite_utils:subscribe(Node, micro_block_created),
     StartRes = rpc:call(Node, aec_conductor, start_mining, [], 5000),
     ct:log("aec_conductor:start_mining() (~p) -> ~p", [Node, StartRes]),
-    Res = mine_blocks_until_loop(ConditionFun, Max),
+    Res = mine_blocks_until_txs_on_chain_loop(Node, TxHashes, Max, []),
     StopRes = rpc:call(Node, aec_conductor, stop_mining, [], 5000),
     ct:log("aec_conductor:stop_mining() (~p) -> ~p", [Node, StopRes]),
     aecore_suite_utils:unsubscribe(Node, block_created),
-    aecore_suite_utils:unsubscribe(Node, micro_block_created),
     case Res of
         {ok, BlocksReverse} ->
             {ok, lists:reverse(BlocksReverse)};
         {error, Reason} ->
             erlang:error(Reason)
     end.
 
-mine_blocks_until_loop(ConditionFun, Max) ->
-    mine_blocks_until_loop(ConditionFun, Max, []).
-
-mine_blocks_until_loop(_ConditionFun, 0,_Acc) ->
+mine_blocks_until_txs_on_chain_loop(_Node, _TxHashes, 0, _Acc) ->
     {error, max_reached};
-mine_blocks_until_loop(ConditionFun, Max, Acc) ->
+mine_blocks_until_txs_on_chain_loop(Node, TxHashes, Max, Acc) ->
     case mine_blocks_loop(1, key) of
-        {ok, Blocks} ->
-            NewAcc = Blocks ++ Acc,
-            case ConditionFun() of
-                true  -> {ok, NewAcc};
-                false ->
-                    mine_blocks_until_loop(ConditionFun, Max - 1, NewAcc)
+        {ok, [Block]} -> %% We are only observing key blocks
+            NewAcc = [Block | Acc],
+            case txs_not_on_chain(Node, Block, TxHashes) of
+                []        -> {ok, NewAcc};
+                TxHashes1 -> mine_blocks_until_txs_on_chain_loop(Node, TxHashes1, Max - 1, NewAcc)
             end;
         {error, _} = Error -> Error
     end.
 
+txs_not_on_chain(Node, Block, TxHashes) ->
+    {ok, BlockHash} = aec_blocks:hash_internal_representation(Block),
+    case rpc:call(Node, aec_chain, get_prev_generation, [BlockHash]) of
```

Comment:
Not sure why using RPC rather than using user API as in [other test](https://github.com/aeternity/epoch/blob/d90447069f135d08a5ef2ca0544c4aed148a3982/apps/aehttp/test/aehttp_integration_SUITE.erl#L6019-L6027) though seems good enough.

Reply:
Everything in aecore_suite_utils uses RPC would be really strange to start using HTTP there?!